### PR TITLE
zephyr: fix compilation when no CONFIG_BOOT_ENCRYPTION_KEY_FILE

### LINKED ISSUE #1297: MCUboot v1.9.0-rc1 fails to build with Zephyr
Building MCUboot v1.9.0-rc1 with Zephyr fails early in the build process
```
~/work/zephyrproject/mcuboot/boot/zephyr/build$ git describe  --dirty
v1.9.0-rc1
~/work/zephyrproject/mcuboot/boot/zephyr/build$ cmake -GNinja -DBOARD=efr32_radio_brd4104a ..
Including boilerplate (Zephyr workspace): ~/work/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: ~/work/zephyrproject/mcuboot/boot/zephyr
-- Zephyr version: 3.0.0-rc1 (~/work/zephyrproject/zephyr), build: v3.0.0-rc1-4-g3a979bb81904
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.10") found components: Interpreter 
-- Found west (found suitable version "0.11.0", minimum required is "0.7.1")
-- Board: efr32_radio_brd4104a
-- Cache files will be written to: ~/.cache/zephyr
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.13.1 (~/.local/opt/zephyr-sdk-0.13.1)
-- Found dtc: ~/.local/opt/zephyr-sdk-0.13.1/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6")
-- Found toolchain: zephyr 0.13.1 (~/.local/opt/zephyr-sdk-0.13.1)
-- Found BOARD.dts: ~/work/zephyrproject/zephyr/boards/arm/efr32_radio/efr32_radio_brd4104a.dts
-- Found devicetree overlay: ~/work/zephyrproject/mcuboot/boot/zephyr/dts.overlay
-- Generated zephyr.dts: ~/work/zephyrproject/mcuboot/boot/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: ~/work/zephyrproject/mcuboot/boot/zephyr/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: ~/work/zephyrproject/mcuboot/boot/zephyr/build/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: ~/work/zephyrproject/mcuboot/boot/zephyr/build/zephyr/dts.cmake
Parsing ~/work/zephyrproject/mcuboot/boot/zephyr/Kconfig
Loaded configuration '~/work/zephyrproject/zephyr/boards/arm/efr32_radio/efr32_radio_brd4104a_defconfig'
Merged configuration '~/work/zephyrproject/mcuboot/boot/zephyr/prj.conf'
Configuration saved to '~/work/zephyrproject/mcuboot/boot/zephyr/build/zephyr/.config'
Kconfig header saved to '~/work/zephyrproject/mcuboot/boot/zephyr/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 10.3.0
-- The CXX compiler identification is GNU 10.3.0
-- The ASM compiler identification is GNU
-- Found assembler: ~/.local/opt/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
MCUBoot bootloader key file: ~/work/zephyrproject/mcuboot/root-rsa-2048.pem
CMake Error at CMakeLists.txt:312 (string):
  string sub-command FIND requires 3 or 4 parameters.
```

The issue was introduced by #1255. The PR erroneously assumed that `CONFIG_BOOT_ENCRYPTION_KEY_FILE` will always be defined.

### PR DESCRIPTION
**Pull request by wcappelle**
Fix compilation of zephyr builds when there is no CONFIG_BOOT_ENCRYPTION_KEY_FILE set.

Fixes #1297

### CODE REVIEW

**Comment by wcappelle** [comment]
to reviewers: this is a pr to the main, should one be created to rc-1 branch as well?

**Review by utzig — APPROVED** [review]
LGTM

File: boot/zephyr/CMakeLists.txt [hunk 1]
```diff
@@ -305,22 +305,23 @@ if(NOT CONFIG_BOOT_SIGNATURE_KEY_FILE STREQUAL "")
...
-  file(READ ${filepath} temp_text)
-  string(FIND "${temp_text}" ${CONFIG_BOOT_ENCRYPTION_KEY_FILE} match)
-  if(${match} GREATER_EQUAL 0)
-    if(NOT DEFINED CONF_DIR)
-      get_filename_component(CONF_DIR ${filepath} DIRECTORY)
-    else()
-      message(FATAL_ERROR "Encryption key file defined in multiple conf files")
-    endif()
-  endif()
-endforeach()
```
Comment (mnkp) [inline boot/zephyr/CMakeLists.txt:309]: Let's maybe remove extra empty lines since they make overall layout inconsistent.

**Review by mnkp — COMMENTED** [review]
LGTM with a minor comment.

File: boot/zephyr/CMakeLists.txt (hunk 1 above)
Comment (wcappelle) [reply boot/zephyr/CMakeLists.txt]: true, done

**Review by mnkp — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**